apps/api/src/app/notifications/usecases/get-activity-stats/get-activity-stats.usecase.ts

```diff
@@ -1,42 +1,61 @@
 import { Injectable } from '@nestjs/common';
 import { NotificationRepository } from '@novu/dal';
-import { subMonths, subWeeks, subYears } from 'date-fns';
-import { ActivityStatsResponseDto } from '../../dtos/activity-stats-response.dto';
+import { isBefore, parseISO, subMonths, subWeeks, subYears } from 'date-fns';
+
 import { GetActivityStatsCommand } from './get-activity-stats.command';
 
+import { ActivityStatsResponseDto } from '../../dtos/activity-stats-response.dto';
+
 @Injectable()
 export class GetActivityStats {
   constructor(private notificationRepository: NotificationRepository) {}
 
   async execute(command: GetActivityStatsCommand): Promise<ActivityStatsResponseDto> {
-    const yearly = await this.notificationRepository.count({
-      _environmentId: command.environmentId,
-      _organizationId: command.organizationId,
-      createdAt: {
-        $gte: subYears(new Date(), 1),
-      },
-    });
+    const now = new Date();
 
-    const monthly = await this.notificationRepository.count({
-      _environmentId: command.environmentId,
-      _organizationId: command.organizationId,
-      createdAt: {
-        $gte: subMonths(new Date(), 1),
+    const yearlyNotifications = await this.notificationRepository.find(
```

Comment:
We retrieve all the notifications in a year and then loop them to calculate the monthly and the weekly rather than hitting the DB again. Also to help for large collections we just retrieve 2 fields: `_id` and `createdAt`. We could even just retrieve `createdAt`.

Reply:
@scopsy this was meant more like a short term fix. I do think we need to review the UI and what's the benefit of showing one year long usage all the time.
But you are right for a long term your proposal should perform better. Let me try to experiment a bit with it in MongoDB.